src/collapse/perf.rs

```diff
@@ -1,123 +1,188 @@
-use super::util::fix_partially_demangled_rust_symbol;
-use super::Collapse;
-use fnv::FnvHashMap;
-use log::warn;
 use std::collections::VecDeque;
-use std::io;
-use std::io::prelude::*;
+use std::io::{self, BufRead};
+
 use symbolic_demangle::demangle;
 
+use crate::collapse::common::{self, CollapsePrivate, Occurrences};
+
 const TIDY_GENERIC: bool = true;
 const TIDY_JAVA: bool = true;
 
-/// Settings that change how frames are named from the incoming stack traces.
-///
-/// All options default to off.
-#[derive(Clone, Debug, Default)]
-pub struct Options {
-    /// Include PID in the root frame.
-    ///
-    /// If disabled, the root frame is given the name of the profiled process.
-    pub include_pid: bool,
+mod logging {
+    use log::{info, warn};
 
-    /// Include TID and PID in the root frame.
-    ///
-    /// Implies `include_pid`.
-    pub include_tid: bool,
+    pub(super) fn filtering_for_events_of_type(ty: &str) {
+        info!("Filtering for events of type: {}", ty);
+    }
 
-    /// Include raw addresses (e.g., `0xbfff0836`) where symbols can't be found.
-    pub include_addrs: bool,
+    pub(super) fn weird_event_line(line: &str) {
+        warn!("Weird event line: {}", line);
+    }
 
-    /// Annotate JIT functions with a `_[j]` suffix.
+    pub(super) fn weird_stack_line(line: &str) {
+        warn!("Weird stack line: {}", line);
+    }
+}
+
+/// `perf` folder configuration options.
+#[derive(Clone, Debug)]
+pub struct Options {
+    /// Annotate JIT functions with a `_[j]` suffix. Default is `false`.
     pub annotate_jit: bool,
 
-    /// Annotate kernel functions with a `_[k]` suffix.
+    /// Annotate kernel functions with a `_[k]` suffix. Default is `false`.
```

Comment:
I wonder if the defaults should be on their own line?

Reply:
Which one do you like better? I'm indifferent.

![image](https://user-images.githubusercontent.com/10109972/61746444-51f5b180-ad61-11e9-9017-427f8685c2d3.png)

Reply:
The top is growing on me, which was your suggested change; so I'll go ahead and make it. If you change your mind, just let me know.